WordPressKitTests/NSDate+RFC3339Tests.swift

```diff
@@ -0,0 +1,43 @@
+@testable import WordPressKit
+import XCTest
+
+// This is an incomplete test for implementing RFC 3339.
+// It's purpose is to ensure our code "works".
+//
+// See also:
+//
+// - https://datatracker.ietf.org/doc/html/rfc3339
+class NSDateRFC3339Tests: XCTestCase {
+
+    func testDateFormatterConfiguration() throws {
+        let rfc3339Formatter = try XCTUnwrap(NSDate.rfc3339Formatter)
+
+        XCTAssertEqual(rfc3339Formatter.timeZone, TimeZone(secondsFromGMT: 0))
+        XCTAssertEqual(rfc3339Formatter.locale, Locale(identifier: "en_US_POSIX"))
+        XCTAssertEqual(rfc3339Formatter.dateFormat, "yyyy'-'MM'-'dd'T'HH':'mm':'ssZ")
+    }
+
+    func testValidRFC3339DateFromString() {
+        XCTAssertEqual(
+            NSDate.with(wordPressComJSONString: "2023-03-19T15:00:00Z"),
+            Date(timeIntervalSince1970: 1_679_238_000)
+        )
+    }
+
+    func testInvalidRFC3339DateFromString() {
+        XCTAssertNil(NSDate.with(wordPressComJSONString: "2024-01-01"))
+    }
+
+    func testInvalidDateFromString() {
+        XCTAssertNil(NSDate.with(wordPressComJSONString: "not a date"))
+    }
+
+    func testValidRFC3339StringFromDate() {
+        XCTAssertEqual(
+            NSDate(timeIntervalSince1970: 1_679_238_000).wordPressComJSONString(),
+            // Apparently, NSDateFormatter doesn't offer a way to specify Z vs +0000.
+            // This might go all the way back to the ISO 8601 and RFC 3339 specs overlap.
+            "2023-03-19T15:00:00+0000"
```

Comment:
Have you tried to put a single quote `'` around the `Z` in the format: `"yyyy'-'MM'-'dd'T'HH':'mm':'ss'Z'"`?

Reply:
BTW, this is one of the reasons I think we should avoid using standard names in function names. It's pretty difficult to ensure conformance.

Reply:
If we are being really strict here, this test case should fail, because the output is not in [ RFC3339 time format](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6).

Reply:
Thanks @crazytonyli !

I didn't noticed that 🤦‍♂️ 

I tried locally and it does the job at returning `2023-03-19T15:00:00Z` as I originally expected. 

But ☝️ other tests that deal with API and dates fail. I think maybe we should leave it like this for the moment, and update / investigate later on. After all, the format the code uses right now is the "original" one.

What do you think?